src/aspire/basis/steerable.py

```diff
@@ -132,10 +142,77 @@ def calculate_bispectrum(
 
         return B
 
-    def rotate(self, complex_coef, radians, refl=None):
+    def rotate(self, coef, radians, refl=None):
+        """
+        Returns coefs rotated by `radians`.
+
+        :param coef: Basis coefs.
+        :param radians: Rotation in radians.
+        :param refl: Optional reflect image (bool)
+        :return: rotated coefs.
+        """
+        # Covert radians to a broadcastable shape
+        if isinstance(radians, np.ndarray):
+            if len(radians) != len(coef):
+                raise RuntimeError(
+                    "`rotate` call `radians` length cannot broadcast with"
+                    f" `coef` {len(coef)} != {len(radians)}"
+                )
+            radians = radians.reshape(-1, 1)
+        # else: radians can be a constant
+
+        # self.angular_indices are `ks`
+        assert len(self.angular_indices) == coef.shape[-1]
+
+        # For all coef in stack,
+        #   compute the ks * radian used in the trig functions
+        ks_rad = np.atleast_2d(self.angular_indices * -1 * radians)
+        ks_pos = ks_rad[:, self.pos_angular_inds]
+        ks_neg = ks_rad[:, self.neg_angular_inds]
+
+        # Slice the coef on postive and negative ells
+        coef_zer = coef[:, self.zero_angular_inds]
+        coef_pos = coef[:, self.pos_angular_inds]
+        coef_neg = coef[:, self.neg_angular_inds]
+
+        # Handle zero case and avoid mutating the original array
+        coef = np.empty_like(coef)
+        coef[:, self.zero_angular_inds] = coef_zer
+
+        # refl
+        if refl is not None:
+            if isinstance(refl, np.ndarray):
+                assert len(refl) == len(coef)
+            # else: refl can be a constant
+            # negate the coefs corresponding to negative ells
+            coef_neg[refl] = coef_neg[refl] * -1
+
+        # Apply formula
```

Comment:
If you take the suggestion above (get rid of `-1` in computing `ks_rad`) the signs in front of the two `sin` functions here will just be flipped i.e.

```
        coef[:, self.pos_angular_inds] = coef_pos * np.cos(ks_pos) - coef_neg * np.sin(
            ks_neg
        )
        coef[:, self.neg_angular_inds] = coef_neg * np.cos(ks_neg) + coef_pos * np.sin(
            ks_pos
        )
```

Reply:
That's the formula you get if you derive starting with the complex rotation formula

```
a_kq' = exp(-i k theta) * a_kq
```

Reply:
Sure, I removed the `-1` * radians without changing the formula to bring us inline with CCW rotation convention as demonstrated by the rotation test comparison with numpy rot90 in`tests/test_FFBbasis2D.py`.

I also slightly changed some comments to reflect this change, and very minor code changes that I think should express some clearer intent.  I just want to review the tests and docs before I push it up.  There is one area I might be able to clean up a bit more.

Its still not clear to me why the complex rotation for `exp(-i k theta)` is going CW, but I have negated that one.